nornir_nautobot/plugins/tasks/dispatcher/default.py

```diff
@@ -499,3 +499,55 @@ def get_config(
             with open(backup_file, "w", encoding="utf8") as filehandler:
                 filehandler.write(running_config)
         return Result(host=task.host, result={"config": running_config})
+
+    @classmethod
+    def merge_config(
+        cls,
+        task: Task,
+        logger,
+        obj,
+        config: str,
+    ) -> Result:
+        """Send configuration to merge on the device.
+
+        Args:
+            task (Task): Nornir Task.
+            logger (logging.Logger): Logger that may be a Nautobot Jobs or Python logger.
+            obj (Device): A Nautobot Device Django ORM object instance.
+            config (str): The config set.
+
+        Raises:
+            NornirNautobotException: Authentication error.
+            NornirNautobotException: Timeout error.
+            NornirNautobotException: Other exception.
+
+        Returns:
+            Result: Nornir Result object with a dict as a result containing what changed and the result of the push.
+        """
+        logger.info("Config merge via netmiko starting", extra={"object": obj})
+        # Sending None to napalm_configure for revert_in will disable it, so we don't want a default value.
+
+        try:
+            push_result = task.run(
+                task=netmiko_send_config,
+                config_commands=config.splitlines(),
+                enable=True,
+            )
+        except NornirSubTaskError as exc:
```

Comment:
Can we add the same error checking on netmiko?

```python
        except NornirSubTaskError as exc:
            if isinstance(exc.result.exception, NetmikoAuthenticationException):
                error_msg = f"`E1017:` Failed with an authentication issue: `{exc.result.exception}`"
                logger.error(error_msg, extra={"object": obj})
                raise NornirNautobotException(error_msg)
            if isinstance(exc.result.exception, NetmikoTimeoutException):
                error_msg = f"`E1018:` Failed with a timeout issue. `{exc.result.exception}`"
                logger.error(error_msg, extra={"object": obj})
                raise NornirNautobotException(error_msg)
            error_msg = f"`E1016:` Failed with an unknown issue. `{exc.result.exception}`"
            logger.error(error_msg, extra={"object": obj})
            raise NornirNautobotException(error_msg)
        if result[0].failed:
            return result
        running_config = result[0].result
        # Primarily seen in Cisco devices.
        if "ERROR: % Invalid input detected at" in running_config:
            error_msg = "`E1019:` Discovered `ERROR: % Invalid input detected at` in the output"
            logger.error(error_msg, extra={"object": obj})
            raise NornirNautobotException(error_msg)
```

Reply:
done